src/extension.ts

```diff
@@ -168,224 +168,222 @@ export function activate(context: ExtensionContext) {
+              }
+            }));
+
+            const api: XMLExtensionApi = {
+              // add API set catalogs to internal memory
+              addXMLCatalogs: (catalogs: string[]) => {
+                const externalXmlCatalogs = externalXmlSettings.xmlCatalogs;
+                catalogs.forEach(element => {
+                  if (!externalXmlCatalogs.includes(element)) {
+                    externalXmlCatalogs.push(element);
+                  }
+                });
+                languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: getXMLSettings(requirements.java_home) });
+                onConfigurationChange();
+              },
+              // remove API set catalogs to internal memory
+              removeXMLCatalogs: (catalogs: string[]) => {
+                catalogs.forEach(element => {
+                  const externalXmlCatalogs = externalXmlSettings.xmlCatalogs;
+                  if (externalXmlCatalogs.includes(element)) {
+                    const itemIndex = externalXmlCatalogs.indexOf(element);
+                    externalXmlCatalogs.splice(itemIndex, 1);
+                  }
+                });
+                languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: getXMLSettings(requirements.java_home) });
+                onConfigurationChange();
+              },
+              // add API set fileAssociations to internal memory
+              addXMLFileAssociations: (fileAssociations: XMLFileAssociation[]) => {
+                const externalfileAssociations = externalXmlSettings.xmlFileAssociations;
+                fileAssociations.forEach(element => {
+                  if (!externalfileAssociations.some(fileAssociation => fileAssociation.systemId === element.systemId)) {
+                    externalfileAssociations.push(element);
+                  }
+                });
+                languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: getXMLSettings(requirements.java_home) });
+                onConfigurationChange();
+              },
+              // remove API set fileAssociations to internal memory
+              removeXMLFileAssociations: (fileAssociations: XMLFileAssociation[]) => {
+                const externalfileAssociations = externalXmlSettings.xmlFileAssociations;
+                fileAssociations.forEach(element => {
+                  const itemIndex = externalfileAssociations.findIndex(fileAssociation => fileAssociation.systemId === element.systemId) //returns -1 if item not found
+                  if (itemIndex > -1) {
+                    externalfileAssociations.splice(itemIndex, 1);
+                  }
+                });
+                languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: getXMLSettings(requirements.java_home) });
+                onConfigurationChange();
+              }
+            };
+            return api;
           });
```

Comment:
Need obvious failure message if the binary fails (for instance if the macOS binary is run on Linux).